### PR TITLE
Encode using text codec when sending message

### PR DESCRIPTION
**Pull request by richardhuaaa**
We need to be able to construct the [`EncodedContent` protobuf](https://github.com/xmtp/proto/blob/xmtpv3/proto/message_contents/content.proto#L30) when sending a message, which will allow us to add support for non-text formats in the future.

This PR implements a text codec, similar to the [pattern used in xmtp-js](https://github.com/xmtp/xmtp-js/blob/main/src/codecs/Text.ts#L17).

### CODE REVIEW

File: xmtp/src/codecs/text.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            authority_id: TextCodec::AUTHORITY_ID.to_string(),
+            type_id: TextCodec::TYPE_ID.to_string(),
+            version_major: 1,
+            version_minor: 0,
+        }
+    }
+
+    fn encode(text: String) -> Result<EncodedContent, CodecError> {
+        Ok(EncodedContent {
+            r#type: Some(TextCodec::content_type()),
```
Comment (jazzz) [inline xmtp/src/codecs/text.rs:25]: Am I correct that a raw string literal is used here because type is a keyword?

Comment (jazzz) [reply xmtp/src/codecs/text.rs:25]: Found my own answer. This is not a raw string literal, as the grammar does not match. This is a "keyword Identifier"

File: xmtp/src/codecs/text.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    const AUTHORITY_ID: &str = "xmtp.org";
+    const TYPE_ID: &str = "text";
+    const ENCODING_KEY: &str = "encoding";
+    const ENCODING_UTF8: &str = "UTF-8";
+}
+
+impl ContentCodec<String> for TextCodec {
+    fn content_type() -> ContentTypeId {
+        ContentTypeId {
+            authority_id: TextCodec::AUTHORITY_ID.to_string(),
```
Comment (jazzz) [inline xmtp/src/codecs/text.rs:16]: If all content types need to be defined at compile time then it doesn't seem like Strings are needed in this case. Types could be changed to use `&'static str`

**Review by jazzz — APPROVED** [review]

**Comment by neekolas** [comment]
@richardhuaaa @jazzz 

Are we sure we need to handle content encoding in Rust? Given we already have codecs written in Swift, Kotlin, and JS, I wonder if we really just want to take in `EncodedContent` as bytes and leave it up to the platform SDKs to handle content encoding/decoding. It doesn't seem strictly necessary to have that in scope for this project.

For the fancier formats (like remote attachments), even if we implemented fully in Rust there would have to be a lot of wrapper code in the native languages anyways to enable creation.

We could still keep this implementation in for the CLI and testing.

**Comment by richardhuaaa** [comment]
@neekolas That's a good point! We should reduce the scope in Rust today and only increase over time as we gain confidence. Maybe conversation sending at the Rust level should just take in bytes for now. I'm on board, and will refactor the codecs out later.

**Comment by neekolas** [comment]
We can keep the `EncodedContent` and `TextCodec` stuff somewhere in the repo, since it's useful for the CLI and tests.  As long as it's not part of the `send_message` flow, callers can choose not to use it if they are calling from the context of a full-featured SDK like iOS or Android.

**Pull request merged**